### PR TITLE
Head stabilization complex terrain & improved visual tracking algorithm

### PR DESCRIPTION
**Pull request by sibocw**
### Description
This is a continuation of #136. Main tasks are:
1. Improve visual tracking algorithm after the track-cam bug fix (WIP)
2. Deploy head stabilization model in closed simulation loop (done)

### Types of changes

- [ ] Bugfix
- [x] New feature
- [ ] Refactor / Code style update (no logical changes)
- [ ] Build / CI changes
- [ ] Documentation Update
- [ ] Other (explain)

### Does this address any currently open issues?
#137

### CODE REVIEW

**Comment by sibocw** [comment]
Should be all good now, but after #143 let me rerun everything again tomorrow.

https://github.com/NeLy-EPFL/flygym/assets/23410765/e971f31e-d262-49c9-aa22-4c27eef25cc7


https://github.com/NeLy-EPFL/flygym/assets/23410765/f3888964-706e-4d80-ad5a-cd90bd9ed500

**Comment by sibocw** [comment]
@tkclam Would you mind taking a look at this? #143 has been merged into this (first merged into dev-v1.0.0, which was then merged into this).

**Comment by tkclam** [comment]
Data collection and training are quite time-consuming. Should we store trained model weights in `flygym/data`?
Also, should we include the required packages in `extras_require` in `setup.py`?
```python
"examples": [
    "lightning",
    "pandas",
    "scikit-learn",
    "seaborn",
    "torch",
    "tensorboardX",
    "flyvision @ https://github.com/TuragaLab/flyvis/archive/refs/heads/main.zip",
]
```

File: flygym/examples/head_stabilization/train_proprioception_model.py [hunk 1]
```diff
@@ -5,96 +5,161 @@
...
+        "ThC_pitch": "Coxa",
+        "ThC_roll": "Coxa_roll",
+        "ThC_yaw": "Coxa_yaw",
+        "CTr_pitch": "Femur",
+        "CTr_roll": "Femur_roll",
+        "FTi_pitch": "Tibia",
+        "TiTa_pitch": "Tarsus1",
+    }
+    dof_subset = [_dof_name_lookup[dof] for dof in dof_subset]
+    mask = [dof in dof_subset for dof in flygym.preprogrammed.all_leg_dofs]
```
Comment (tkclam) [inline flygym/examples/head_stabilization/train_proprioception_model.py:35]: value unused

**Comment by sibocw** [comment]
> Data collection and training are quite time-consuming. Should we store trained model weights in `flygym/data`? Also, should we include the required packages in `extras_require` in `setup.py`?
> 
> ```python
> "examples": [
>     "lightning",
>     "pandas",
>     "scikit-learn",
>     "seaborn",
>     "torch",
>     "tensorboardX",
>     "flyvision @ https://github.com/TuragaLab/flyvis/archive/refs/heads/main.zip",
> ]
> ```

Yup sounds good. About `tensorboard`/`tensorboardX` though, I think it's shipped with `lightning`. Maybe it's better if we let `lightning` take care of installing the correct version/variant?

**Comment by tkclam** [comment]
> > Data collection and training are quite time-consuming. Should we store trained model weights in `flygym/data`? Also, should we include the required packages in `extras_require` in `setup.py`?
> > ```python
> > "examples": [
> >     "lightning",
> >     "pandas",
> >     "scikit-learn",
> >     "seaborn",
> >     "torch",
> >     "tensorboardX",
> >     "flyvision @ https://github.com/TuragaLab/flyvis/archive/refs/heads/main.zip",
> > ]
> > ```
> 
> Yup sounds good. About `tensorboard`/`tensorboardX` though, I think it's shipped with `lightning`. Maybe it's better if we let `lightning` take care of installing the correct version/variant?
> 
> Edit: ~I will also add `networkx` since it's used in another example.~ `networkx` was previously included under an entry of `extras_require` called `tutorials`. I will move it to `examples` too.

Just tested the installation of `lightning` with a new environmen. Looks like `tensorboardX` is only included if we install it with `pip install "lightning[extra]"`

**Comment by sibocw** [comment]
> > > Data collection and training are quite time-consuming. Should we store trained model weights in `flygym/data`? Also, should we include the required packages in `extras_require` in `setup.py`?
> > > ```python
> > > "examples": [
> > >     "lightning",
> > >     "pandas",
> > >     "scikit-learn",
> > >     "seaborn",
> > >     "torch",
> > >     "tensorboardX",
> > >     "flyvision @ https://github.com/TuragaLab/flyvis/archive/refs/heads/main.zip",
> > > ]
> > > ```
> > 
> > 
> > Yup sounds good. About `tensorboard`/`tensorboardX` though, I think it's shipped with `lightning`. Maybe it's better if we let `lightning` take care of installing the correct version/variant?
> > Edit: ~I will also add `networkx` since it's used in another example.~ `networkx` was previously included under an entry of `extras_require` called `tutorials`. I will move it to `examples` too.
> 
> Just tested the installation of `lightning` with a new environmen. Looks like `tensorboardX` is only included if we install it with `pip install "lightning[extra]"`

ok then i will include tensorboardX explicitly. Thanks!

**Comment by sibocw** [comment]
All issues should be addressed now. I don't think the change broke anything but I will rerun all the data collection / training / closed look simulation again to make extra sure.

**Review by tkclam — APPROVED** [review]

**Pull request merged**